app/controllers/chat_controller.rb

```diff
@@ -278,6 +278,16 @@ def set_user_last_read
       .where(user: current_user, chat_channel: @chat_channel)
       .update_all(last_read_message_id: params[:message_id])
 
+    mention_notifications = Notification
+      .where(user: current_user)
+      .where(notification_type: Notification.types[:chat_mention])
+      .where(read: false)
+      .where("data LIKE ?", "%\"chat_channel_id\":#{@chat_channel.id}%")
+
+    mention_notifications.each do |mention|
+      mention.update(read: true)
```

Comment:
```suggestion
      mention.update!(read: true)
```

(unless there's a reason to use the non-exploding call 😃)

Reply:
…actually, if there's a chance of many records in `mention_notifications` maybe doing the update in a single query would be better?

Reply:
I actually do want this to be a non-exploding call, and a failure there would break all of chat state tracking (well it just wouldn't publish so you'd have to refresh)